Lib/test/test_ctypes/test_refcounts.py

```diff
@@ -97,5 +97,20 @@ def func(a, b):
         f(1, 2)
         self.assertEqual(sys.getrefcount(ctypes.c_int), a)
 
+    @support.refcount_test
+    def test_callback_py_object_none_return(self):
+        # bpo-36880: test that returning None from a py_object callback
+        # does not decrement the refcount of None.
+
+        for FUNCTYPE in (ctypes.CFUNCTYPE, ctypes.PYFUNCTYPE):
+            with self.subTest(FUNCTYPE=FUNCTYPE):
+                @FUNCTYPE(ctypes.py_object)
+                def func():
+                    return None
+
+                # Check that calling func does not affect None's refcount.
+                for _ in range(10000):
```

Comment:
Single call should be enough to trigger the refleak checker.

Reply:
Unfortunately not... I tested again - apparently the refleak checker (with default settings: `-m test --huntrleaks=: test.test_ctypes.test_refcounts`) only detects if `None`'s refcount is too *high*, not if it's too low. The latter case is only detected once `None`'s refcount gets to zero/negative (which triggers an assertion), and that requires a thousand or so iterations of the broken code.

Reply:
If `None` refcount goes too small it will crash at some point assuming the rest of the code decrements normally

Reply:
> Unfortunately not... I tested again - apparently the refleak checker (with default settings: -m test --huntrleaks=: test.test_ctypes.test_refcounts) only detects if None's refcount is too high, not if it's too low. The latter case is only detected once None's refcount gets to zero/negative (which triggers an assertion), and that requires a thousand or so iterations of the broken code.

I see, it is because of https://github.com/python/cpython/issues/74959.